Tribler/Test/Core/Modules/RestApi/test_metadata_endpoint.py

```diff
@@ -123,6 +124,30 @@ def test_subscribe(self):
         return self.do_request('metadata/channels/%s' % channel_pk, expected_code=200,
                                request_type='POST', post_data=post_params)
 
+    def test_unsubscribe(self):
+        with db_session:
+            channel = self.session.lm.mds.ChannelMetadata.select()[:][0]
+            channel.subscribed = True
+
+        self.should_check_equality = False
+        post_params = {'subscribe': '0'}
+        channel_pk = hexlify(channel.public_key)
+
+        def on_response(_):
+            # stupid workaround for polling results of a background process
```

Comment:
I don't get why this complicated logic (with a sleep!) is necessary?

Reply:
https://github.com/Tribler/tribler/blob/0149098456ae1cbba4d44ead3e0c2ccb6ec08c4d/Tribler/Core/Modules/restapi/metadata_endpoint.py#L171

The deletion is handled by a background thread without `Deferred` attached to it. So, I have to poll the database constantly to check if it is finished.

Reply:
Right. Is this the only location where you have this kind of check? Otherwise, it might be a good idea to extract this code and make a generic method out of it.

Also, please use asynchronous sleeps instead of `time.sleep`.